### PR TITLE
Added UART2 support for Radio

### PR DESCRIPTION
**Pull request by gunturiCM**
# Main Changes
Added UART2 support for radio.
Now we can use UART2 on ESP32 to read radio data from Raspberry Pi. This will free UART0 for programming easily.

## About code changes
1. Added `beginSerial` function to the `radio.hpp`.
2. Modified `main.cpp` to call `beginSerial`.
3. Modified `readPiData` in `radio.cpp`. Check the file for details.
4. Modified `config.hpp`.

## Usage
Inside `config.cpp`, uncomment `USE_UART2_FOR_RADIO` macro to use UART2 for radio communication. Comment it to use UART0 instead. Check the macros `RXD2` and `TXD2` inside `config.hpp` if they have the correct pins defined.
**Note:** You need to change physical UART connections to the appropriate pins accordingly.

### CODE REVIEW

File: include/config.hpp [hunk 1]
```diff
@@ -13,6 +13,18 @@
 /*
  * Serial config
  */
+// uncomment the below line to use a seperate UART port for radio communication
+// this frees UART0 and the USB port for programming
+#define USE_UART2_FOR_RADIO  // UART2 is readily useable, UART1 requires more configuration as it shares pins with SPI
+#ifdef USE_UART2_FOR_RADIO
+    #include <HardwareSerial.h>
```
Comment (alphaville) [inline include/config.hpp:20]: Don't mix `#define`s and `#include`s in the code. Move the `#include` to the top

File: include/config.hpp [hunk 2]
```diff
@@ -13,6 +13,18 @@
 /*
  * Serial config
  */
+// uncomment the below line to use a seperate UART port for radio communication
+// this frees UART0 and the USB port for programming
+#define USE_UART2_FOR_RADIO  // UART2 is readily useable, UART1 requires more configuration as it shares pins with SPI
```
Comment (alphaville) [inline include/config.hpp:18]: Is there a reason for this? We can simply move to UART2.

File: include/config.hpp (hunk 1 above)
Comment (gunturiCM) [reply include/config.hpp:20]: On it. Thank you.

File: src/main.cpp [hunk 3]
```diff
@@ -16,6 +16,7 @@ long int val; // variable to read the value from the analog pin
 void setup()
 {
   Serial.begin(115200);
+  radio.beginSerial(115200);
```
Comment (alphaville) [inline src/main.cpp:19]: It seems that `Serial` is not used anywhere, right? We should remove it.

**Review by alphaville — CHANGES_REQUESTED** [review]

File: include/config.hpp (hunk 2 above)
Comment (gunturiCM) [reply include/config.hpp]: Modified. Now using UART2 for radio by default.

File: src/main.cpp (hunk 3 above)
Comment (gunturiCM) [reply src/main.cpp:18]: Removed `Serial.begin`.

File: include/radio.hpp [hunk 4]
```diff
@@ -42,6 +42,11 @@ namespace bzzz
             
             Radio();
 
+            /**
+             * @brief Begin Serial com
+            */
+           void beginSerial(int baud);
```
Comment (alphaville) [inline include/radio.hpp:48]: Minor comment: just add `@param baud baud rate`

Comment (gunturiCM) [reply include/radio.hpp:48]: On it.

**Review by alphaville — CHANGES_REQUESTED** [review]
Overall, it looks good to me, but we should test it on the ESP with RPi and make sure it works

File: include/radio.hpp (hunk 4 above)
Comment (gunturiCM) [reply include/radio.hpp:49]: Added `@param baud baud rate` as suggested.

**Comment by jamie-54** [comment]
Hi All, I think we should try this on the hardware before merging. I was trying to do basically the same thing today on the hardware and when I changed everything to serial2 it seemed to work fine, but when I tired using serial2 for Pi to ESP and serial to read the `logSerial`  I ran into issues.

If we are using seiral2 just to keep programming port free it should be fine as long as we aren't trying to use the `logSerial` function as it uses serial.

**Comment by alphaville** [comment]
I think we should close this for now as we experienced some strange behaviour with UART2. We can of course re-open it in the future.

**Pull request closed without merging**